library/core/src/slice/mod.rs

```diff
@@ -2248,6 +2248,10 @@ impl<T> [T] {
     /// let r = s.binary_search_by_key(&1, |&(a, b)| b);
     /// assert!(match r { Ok(1..=4) => true, _ => false, });
     /// ```
+    // Lint rustdoc::broken_intra_doc_links is allowed as `slice::sort_by_key` is
+    // in crate `alloc`, and as such doesn't exists yet when building `core`
+    // links to downstream crate: https://github.com/rust-lang/rust/issues/74481
```

Comment:
This is not the full story: this only works because the documentation for primitives only shows up in libstd, not libcore.
```suggestion
    // links to downstream crate: #74481. Since primitives are only documented in libstd (#73423),
    // this never leads to broken links in practice.
```